### ISSUE TITLE
[Bug]: load data s3 csv file, 20 columns of 100 million(34G), CN OOM .

### ISSUE THREAD

**Issue by tianyahui-python** [matrixorigin/matrixone#8577]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

Reproducible Steps:

in the mo-cloud: 
execute load data s3 csv file, 20 columns 100 million (34G).  CN OOM

CN Configuration:
<img width="1446" alt="截屏2023-03-21 16 43 11" src="https://user-images.githubusercontent.com/39303074/226562332-6989c87c-6f00-4a58-9d8d-838b3064dffa.png">

pod info:
<img width="1728" alt="截屏2023-03-21 16 03 54" src="https://user-images.githubusercontent.com/39303074/226562520-dca5c773-36f2-4703-9363-5ce9db67a3b8.png">

Please talk to me in private if you need detailed reenactment steps.

**Comment by nnsgmsone** [comment]
有日志吗？ @tianyahui-python

**Comment by tianyahui-python** [comment]
> 有日志吗？ @tianyahui-python
Now I can't get the log. @nnsgmsone

**Comment by nnsgmsone** [comment]
重构后，load占用的内存应该会很少

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
关联issue https://github.com/matrixorigin/matrixone/issues/7891

**Comment by nnsgmsone** [comment]
fixed by #9069

**Comment by nnsgmsone** [comment]
fixed by #9069.

**Comment by Ariznawlll** [comment]
![image](https://user-images.githubusercontent.com/108530700/236603317-45cb2c17-af2e-4841-8118-afae57c5360e.png)

test done！
